### ISSUE TITLE
Mobile Bug Management - Q1 2024

### ISSUE THREAD

**Issue by bischoffa** [department-of-veterans-affairs/va-mobile-app#5367]
## Purpose of this feature:
- ONLY for BUG ticket 
- Only bug tickets that have been scrubbed (reviewed by QA indicating they are ready to be reviewed by engineering)
- Prevent a large culmination of tech debt backlog that will force Mobile's app to work on 

## Process:
- [Process document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-

**1.) A bug ticket is created by anyone**
**2.) QA scrubs the bug ticket (adds labels, ensures it includes applicable info (test user, steps to repeat, etc)**
- If bug ticket needs more information QA may assign the ticket to the ticket creator or appropriate Mobile team to get more details before it can progress 
- Once scrubbed successful adds estimate of 1 (amount of time engineering will spend on it initially) and 
    - Add Scrub label to the ticket
    - Add appropriate team label to the ticket
    - Add ticket to the Bug [epic](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/5367)
    - Move ticket to Backlog Ready to Work Column if ready
    - If its blocked by more information, desired state is needed move it to Blocked column 
     - If its a third party bug ticket (bug ticket that has been opened with another team - label will be bug but not tied to the bug epic and managed exclusive by engineering ex [7109](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7109)

**3.) Product Managers on the Specialized Mobile Teams will routinely review their bug backlog and handle any blockers / requirements / desired state work**
- In the case the instance another team is needed to determine the solution a separate ticket will be created for that area for them to do their work and the PM will need to coordinate with appropriate teams for resources / capacity / priority to complete 
   - Example [3051](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/3051)
   - For UX ticket they will have in their name UX Review and also be tagged with Bug and UX labels for finding
   - Once UX Review tickets are done then the original bug ticket is updated with findings and ready for engineering work to be prioritized 

**4.) Product Managers on the Specialized Mobile Team prioritizes their own relevant bug tickets within their team as needed**
- During prioritization PM on Mobile teams will need to coordinate with QA, if QA resources are needed to validate, if QA has capacity to test
- If PM thinks that the ticket should be done by another team then its that PM's responsibility to have that convo; PM will not go back to QA to reassess owner

**5.) Bug tickets are then worked in the Sprint** 

**6a.) If Engineering resolves the ticket within point capacity**
- QA test and validates work and confirms issue is no longer a bug 
    - Issues that arise will lead to other bug tickets that may be worked in existing sprint 
- Ticket is ready to be released

**6b.) If ticket is not resolved within estimate, resolution will not happen that sprint** 
- Engineering will add artifacts / information to the ticket such as - their findings, solution to resolve, are other teams are needed,  if its resolvable, if not resolvable a work around, and other applicable info for when the ticket is considered in a future sprint 
- Engineer will update the ticket sizing to reflect their finding and anticipated solution 
- Engineering will move the ticket to the Backlog Ready to Work for Product to consider for prioritization in the future 

**7.)Product includes this ticket back in their Step 4 and repeat**

## Historical and Notes:
- Currently in Q2 planning there is not an allotted capacity for bugs so will pull them in when capacity allows (pull vs push method)
- Bugs associated with ongoing development of a project / feature will be included in the Sprint as part of that project's capacity thus not treated the same here 
- When bugs are created they are assigned 1 point, purpose is for engineering to spend 1 point looking into it and then if another ticket is needed a ticket needs to be spun off then sized. 
- When input from another practice is needed before Eng could fix (ex: need UI/design from UX team), QA should add a "blocked-[practice]" label during bug scrub (ex: "blocked-UX" for the previous example)
- For Q3 there is a single bucket of capacity being shared for bugs and discipline tech debt - discipline tech debt tickets should NOT be added to this they should use their own epics to track, manage and prioritize their work
- Q4 Planning will assume the same process that was done in Q3
- Starting 12/6 Process is changing to align with the new individual teams Mobile has created - meaning each team will be responsible for their relevant bug backlog, scoping, sizing, prioritizing, and working. These individual teams will also be responsible for coordinating with other teams like QA in ensuring capacity is planned for applicable testing - changes communicated to PMs on 11/21 [Slack thread](https://adhoc.slack.com/archives/C05S1S1MQ3G/p1700577597899419) post Matt approval and then communicated to va-mobile-managers [Slack channel ](https://adhoc.slack.com/archives/C02CTV4JN0J/p1701177813973419)on 11/28

**Comment by bischoffa** [comment]
Backlog of Bug ticket as of Sprint 73

- Sev 1 - 0
- Sev 2 - 8
- Sev 3 - 45
- Sev 4 - 1

**Comment by bischoffa** [comment]
Backlog as of Sprint 74

- Sev 1 - 0
- Sev 2 - 8 (4 in blocked) 
- Sev 3 - 66 (10 in blocked)

Note Sev system changed from Sev 1 -> 4 to Sev 1 -> 3, thus no Sev - 4 reporting

**Comment by bischoffa** [comment]
Backlog as of Sprint 75

- Sev 1 - 0
- Sev 2 - 9
- Sev 3 - 62

**Comment by bischoffa** [comment]
Backlog as of 11/14

- Sev 1 -  0
- Sev 2 - 34 
- Sev 3 - 0 

 Backlog 
- 20 bugs totaling 51 points
- 9 blocked / iceboxed 
- 11 ready to be scheduled (35 points)

**Comment by bischoffa** [comment]
Starting Sprint 86 - individual mobile teams will be managing their own bug backlog - meaning they will monitor new bugs and determine when to prioritize work within their teams. 

https://adhoc.slack.com/archives/C05S1S1MQ3G/p1700577597899419
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/features/Bug%20Managment.md

**Comment by bischoffa** [comment]
Looks like this was used for Q1 2024 - so closing it as we are now in Q2. 

Each individual Mobile team now has their own bug management epic to manage and prioritize work.

**Comment by bischoffa** [comment]
There were tickets leftover in this epic that were not moved to the individual teams epics for q2 bug debt. Moving the tickets now.

**Comment by bischoffa** [comment]
All unclosed tickets have been moved out of this epic to their team specific Q2 bug epic for Global, API, and Health and Benefits

**Comment by bischoffa** [comment]
All unclosed tickets have been moved out of this epic to their team specific Q2 bug epic for Global, API, and Health and Benefits
